packages/taiko-client/driver/chain_syncer/chain_syncer.go

```diff
@@ -46,12 +46,13 @@ func New(
 	p2pSync bool,
 	p2pSyncTimeout time.Duration,
 	blobServerEndpoint *url.URL,
+	latestBlockIDSeenInEventCh chan uint64,
```

Comment:
Is this identical to `headL1Origin`?

Reply:
it updates before headL1Origin because we need to cycle through every single block first to check if its preconfirmed which in a large batch can take some time and sequencers will be able to preconf a block that will just get re-orged/wiped out, but gossiped.